Milestone 3 completion

RNtaate opened the pull request:
## What does this PR do?
- [X] Connects the `CREATE_BOOK` and `REMOVE_BOOK` actions to the component.
- [X] Makes the `BooksForm ` component stateful.

Guy-Gustave committed:
add localState and implement handlechange method

Guy-Gustave committed:
implement adding new book using connect

RNtaate committed:
Implement remove book using connect, create handleRemoveBook function

RNtaate committed:
Update Book component with remove button, use props for onclick functionality

RNtaate committed:
Fix eslint stylelint linter offenses

addod19 commented on src/containers/BooksForm.js:
```diff
@@ -1,16 +1,65 @@
+        {categories.map((el) => <option key={el} value={el}>{el}</option>)}
       </select>
-      <button type="button">Submit</button>
+      <button type="button" onClick={handleSubmit}>Submit</button>
```
Kindly change the `onClick` event handler to `onSubmit` which is more descriptive

addod19 reviewed (changes requested):
## Status: Require changes :recycle: :traffic_light: 

Hi @RNtaate and @Guy-Gustave, congratulations on  this milestone :fireworks: The BookForm, BookList, and the Book component neatly design :slightly_smiling_face: However, we got a few comments to address before approving this milestone: Kindly check the comments below:
- Kindly add a title and a description of the Project in the README file so anybody can actually know what is going on there
- Also, one thing to fix in the BookForm component is to change the `onClick` event to `onSubmit` which is what we intend to do

Once these are resolved, kindly submit another code review request :slightly_smiling_face: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

RNtaate committed:
Switch form button from onClick to form onSubmit, update handleSubmit function

RNtaate committed:
Update readme file

RNtaate commented:
## Changes made
 - [X] Changed `onClick` event handler to `onSubmit` event handler in BooksForm component.
 - [X] Updated readme file with a  project title and description.

uryelah commented on src/components/Book.js:
```diff
@@ -2,22 +2,31 @@ import React from 'react';
 import PropTypes from 'prop-types';
 
 const Book = (props) => {
```
- [ ] [OPTIONAL] You can destruct the props in the declaration line of the component in order to save lines of code:
```javaScript
const Book = ({book, remove}) => {
```

uryelah commented on src/components/Book.js:
```diff
@@ -2,22 +2,31 @@ import React from 'react';
   );
 };
 
 Book.defaultProps = {
   book: {},
+  remove: () => {},
 };
 
 Book.propTypes = {
   book: PropTypes.objectOf(PropTypes.object),
+  remove: PropTypes.func,
 };
```
- [ ] [OPTIONAL] The remove function not being passed as a prop would basically break one of the functionalities of the app, right?
That means it should be a required prop, with no need of a default prop value.

uryelah commented on src/containers/BooksForm.js:
```diff
@@ -1,16 +1,66 @@
+    },
+  );
+
+  const handleChange = (e) => {
+    switch (e.target.name) {
+      case 'input':
+        setLocalBook({ id: localbook.id, title: e.target.value, category: localbook.category });
+        break;
+
+      case 'select':
+        setLocalBook({ id: localbook.id, title: localbook.title, category: e.target.value });
+        break;
+
+      default:
+    }
+  };
```
Interesting implementation you did her with the switch case logic.
Can you think of a why to avoid repeating the setLocalBook call?

uryelah reviewed (approved):
Hi **Roy Ntaate & Guy Nigaba** , I'm Sarah your reviewer for this PR 👩‍🚀 

Great job implementing the requests from the last reviewer!
I specially like your creativity in handling the change events for the inputs.

Your project is approved but I mention some optional changes that can improve your project a little so consider checking them out before merging 😉 

Keep on coding and Rock on! 🎸 👨‍🎤 

# Approved 🚀 🎉 

------------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

Guy-Gustave committed:
destruct the props in the declaration line of the component

Guy-Gustave committed:
Merge branch 'milestone-3' of github.com:Guy-Gustave/Bookstore into milestone-3
